### PR TITLE
implement all? any? and inject for Hash

### PR DESCRIPTION
**Pull request by jreinert**

### CODE REVIEW

File: spec/std/hash_spec.cr [hunk 1]
```diff
@@ -503,4 +503,77 @@ describe "Hash" do
...
+        raise Exception.new("continued iterating") if v == 'd'
+        v == 'a' # this is false for the first kv pair
+      end
+    end
+  end
+
+  describe "any?" do
+    it "passes key and value into block" do
+      hash = {a: 'b'}
+      hash.all? do |k, v|
```
Comment (ysbaddaden) [inline spec/std/hash_spec.cr]: Should be `any?` ?

**Comment by jreinert** [comment]
ah, no of course not... sorry bout that. Will fix it right away.

**Comment by ysbaddaden** [comment]
What about `any?` with no block, to check whether a hash has any key? I know there is already `empty?` but it's sometimes nicer to if `hsh.any?` than a double negation unless `hsh.empty?`

**Comment by jreinert** [comment]
good point. Why not

**Comment by asterite** [comment]
Thanks! And thanks for the extensive specs :-)

**Pull request merged**